JSON to Salesforce - Customers

jannejjj opened the pull request:
The customers are saved as contacts into salesforce since the customer object type is a bit restrictive.

closes #117 

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters.
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots.
- [x] Template parameters checked
  - [x] Parameter descriptions are good and end with dots.
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output
- [x] End "balls" like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Tempalte files are going to the correct folder


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a template process for reading customer information from a JSON file and creating or updating customers in Salesforce as contacts.
  - Added documentation outlining prerequisites, implementation details, usage notes, sample JSON data, and error handling for the new process.
  - Introduced metadata for the new process version 1.0.0.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

FrendsSarlinS reviewed (changes requested):
Ticks most of the boxes, nice work! Some comments:

1) In description, maybe we could emphasize as well that the customers are imported to Salesforce as contacts? It’s highlighted well in the process itself, but I think it should be mentioned here as well.
2) Both insert and update statements have extra newlines in them (Enter).
3) For end-user clarity, the “Get contact information from Salesforce“ could have newlines between query parameters, like this:
SELECT Id, Name
FROM Contact
WHERE Email = ‘{{#var.customer.Email}}’
4) The expressions for incrementing the counter after insert and update checks are not needed, as both branches can have their own returns with #var.updatedCount += 1 and #var.insertedCount += 1.
5) Should we include the variable initializations in a separate group as well, or have a mutual text annotation for all of them to be descriptive to the end-user?
6) Not too sure about this one, but the return ending the process might need a title like the others. Maybe just something like “Success, return a message with counters and possible errors”.

jannejjj commented:
Thanks for the comments, addressed all of them in the new commits. I also changed the name of the folder the process is in to match the naming convention we discussed earlier today!

FrendsSarlinS reviewed (approved):
With these changes, I approve this PR!

FrendsSarlinS commented on Templates/CRMs/Salesforce/JSON to Salesforce - Customers/long-description.md:
```diff
@@ -0,0 +1,69 @@
+This template process reads customer information from a **JSON** file in the local fileshare and creates or updates the customers in **Salesforce** as contacts.
+Using this template, you can insert or update customers into Salesforce as contacts.
```
Add a newline here.

FrendsSarlinS commented on Templates/CRMs/Salesforce/JSON to Salesforce - Customers/long-description.md:
```diff
@@ -0,0 +1,69 @@
+# Implementation and Usage Notes
+
+This template reads a JSON file from the local fileshare containing customer data and creates or updates the customers as contacts into Salesforce.
+Process variables include fields for the Salesforce credentials and the path to the input file.
```
Newline here?

FrendsSarlinS commented on Templates/CRMs/Salesforce/JSON to Salesforce - Customers/long-description.md:
```diff
@@ -0,0 +1,69 @@
+# Error Handling
+
+This template checks for errors after each task, and proceeds accordingly. If an error is encountered when e.g., creating a contact into Salesforce, the process moves on to the next customer and an error message is appended to the error variable which is displayed at the end of the process.
+If transient errors are expected, retries for the file read and Salesforce connections can be configured in the tasks.
```
Add a newline here.
